lib/shelf_packages_handler.dart

```diff
@@ -16,20 +16,16 @@ import 'src/package_config_handler.dart';
 /// A handler that serves the contents of a virtual packages directory.
 ///
 /// This effectively serves `package:${request.url}`. It locates packages using
-/// the package resolution logic defined by [resolver]. If [resolver] isn't
-/// passed, it defaults to the current isolate's package resolution logic.
+/// the package resolution logic defined by [packageConfig]. If [packageConfig]
+/// isn't passed, it defaults to the current isolate's package config.
 ///
 /// This can only serve assets from `file:` URIs.
-Handler packagesHandler({PackageResolver resolver}) {
-  resolver ??= PackageResolver.current;
-  return AsyncHandler(resolver.packageRoot.then((packageRoot) {
-    if (packageRoot != null) {
-      return createStaticHandler(p.fromUri(packageRoot),
-          serveFilesOutsidePath: true);
-    } else {
-      return PackageConfigHandler(resolver);
-    }
-  }));
+Handler packagesHandler({PackageConfig packageConfig}) {
```

Comment:
Do we want to go with `Map<String, Uri>` here too?

We'll still need the dependency to `package_config`, but we are more flexible in the future if we avoid putting it into our public API.

Reply:
The problem with that is we can't go from that format to a `PackageConfig` easily.

Reply:
We could however probably drop the `package_config` dependency entirely and use like `Isolate.resolvePackageUri` with a fake uri to get the package directory instead.